packages/core/src/service/plugin/index.test.js

```diff
@@ -24,9 +24,24 @@ import React from 'react'
 import unexpected from 'unexpected'
 
 import PluginService from './index'
+import { Migration } from './classes';
+
+const FOO = 'foo'
+const WRONG_CONTENT_VERSION = '0.0.0'
```

Comment:
What's the difference between wrong, extra, and valid?

Reply:
Wrong is one less (older version), valid is the one we actually use, extra is one more (newer version). The whole change should only work when in data the version is let's say 0.0.1 and your installed is 0.0.2. If your installed is 0.0.0 it will not work and fallback to plugin missing.

Reply:
The example you gave is just for this specific test case, right? Otherwise we should probably use semver. So `^1.0.0` matches `1.*.*`,  `~1.0.0` matches `1.0.*`  and so on. This would also imply that a plugin has a specific version as well as a "compatibleWithVersion" which would be a semver constraint. This should be refactored.

Also, let's call the constants here `OLDER_VERSION`, `MATCHING_VERSION`, and `NEWER_VERSION`. This makes it much easier to read.

Reply:
Yeah sure, the constants are only for test cases but I can change those. The changes to semver however, I am not sure about those. Tempted to leave those maybe for a different PR cause it feels like it can open a can worms.

Reply:
I don't think it will be incredibly difficult, but it's ok to reserve it for another PR. Then my suggestion would be to just do a strict version match `0.0.1 === 0.0.1`, `0.0.1 !== 0.0.2` and remove the custom interpretation of version ranges. If the content version is smaller than the plugin version `0.0.1 <= 0.0.2`, we run it through migrations and migrations is expected to handle it (or throw an exception if not able to).